batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/dao/StudentDao.java

```diff
@@ -0,0 +1,203 @@
+    List<Student> studentList = new ArrayList<Student>();
+    Connection conn = null;
+    PreparedStatement ps = null;
+    ResultSet studentResult = null;
+    try {
+      conn = DbUtils.getConnection();
+
+      String query = "SELECT * FROM student LIMIT ? OFFSET ?";
+      ps = conn.prepareStatement(query);
+      ps.setInt(1, limit);
+      ps.setInt(2, offset);
+      studentResult = ps.executeQuery();
+
+      while (studentResult.next()) {
+        Integer id = studentResult.getInt("id");
+        Integer rollNo = studentResult.getInt("roll");
+        String department = studentResult.getString("department");
+        String batch = studentResult.getString("batch");
+        String name = studentResult.getString("name");
+        String address = studentResult.getString("address");
+        Date dob = studentResult.getDate("dob");
+
+        Student student = new Student();
+        student.setId(id);
+        student.setRoll(rollNo);
+        student.setDepartment(department);
+        student.setBatch(batch);
+        student.setName(name);
+        student.setAddress(address);
+        student.setDob(dob);
+
+        studentList.add(student);
+      }
+
+      return studentList;
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    } finally {
+      DbUtils.closeResultSet(studentResult);
+      DbUtils.closePreparedStatement(ps);
+      DbUtils.closeConnection(conn);
+    }
+  }
+
+  public void insert(Student stud) throws DataException {
+    Connection conn = null;
+    PreparedStatement ps = null;
+    try {
+      conn = DbUtils.getConnection();
+
+      String studentQuery = "INSERT INTO student (name,address,dob,department,batch,roll) VALUES(?,?,?,?,?,?)";
+      ps = conn.prepareStatement(studentQuery);
+      ps.setString(1, stud.getName());
+      ps.setString(2, stud.getAddress());
+      ps.setDate(3, new java.sql.Date(stud.getDob().getTime()));
+      ps.setString(4, stud.getDepartment());
+      ps.setString(5, stud.getBatch());
+      ps.setInt(6, stud.getRoll());
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    } finally {
+      DbUtils.closePreparedStatement(ps);
+      DbUtils.closeConnection(conn);
+    }
+
+  }
+
+  public void delete(Integer studentId) throws DataException {
+    Connection conn = null;
+    PreparedStatement ps = null;
+    try {
+      conn = DbUtils.getConnection();
+
+      String query = "DELETE FROM student WHERE id=?";
+      ps = conn.prepareStatement(query);
+      ps.setInt(1, studentId);
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    } finally {
+      DbUtils.closePreparedStatement(ps);
+      DbUtils.closeConnection(conn);
```

Comment:
:+1:  for closing resources inside finally. Still the better way is to manage through automatic resource management feature.